### PR TITLE
yolov7 v0.1

### PR DESCRIPTION
**Pull request by lindsayshuo**
yolov7 v0.1

### CODE REVIEW

**Comment by fire717** [comment]
1. your gen_wts.py not read args -w -o but hard encode "yolov7.pt"
2. I run -s yolov7t.wts yolov7t.engine t ,got error `Could not compute dimensions for (Unnamed Layer* 187) [Convolution]_output, because the network is not vali`

**Review by wang-xinyu — CHANGES_REQUESTED** [review]
@lindsayshuo Thanks.

Please remove all the binary files, including .engine and .jpg.
For the sample images, we can create a soft link, like yolov5.

File: yolov7/yololayer.h [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    static constexpr int CHECK_COUNT = 3;
+    static constexpr float IGNORE_THRESH = 0.5f;
+    struct YoloKernel
+    {
+        int width;
+        int height;
+        float anchors[CHECK_COUNT * 2];
+    };
+    static constexpr int MAX_OUTPUT_BBOX_COUNT = 1000;
+    static constexpr int CLASS_NUM = 80; //====�޸�Ϊ�Լ������=====
```
Comment (wang-xinyu) [inline yolov7/yololayer.h:20]: please remove this in-line comment.

File: yolov7/yolov7.cpp [hunk 2]
```diff
@@ -0,0 +1,2491 @@
...
+    if (x == 1) return 1;
+    int r = round(x * gd);
+    if (x * gd - int(x * gd) == 0.5 && (int(x * gd) % 2) == 0) {
+        --r;
+    }
+    return std::max<int>(r, 1);
+}
+
+
+//----------------------------yolov7e6e网络模型结构搭建---------------------------------------
```
Comment (wang-xinyu) [inline yolov7/yolov7.cpp:44]: Chinese

File: yolov7/common.hpp [hunk 3]
```diff
@@ -0,0 +1,683 @@
...
+#include <sstream>
+#include <vector>
+#include <opencv2/opencv.hpp>
+#include "NvInfer.h"
+#include "yololayer.h"
+using namespace nvinfer1;
+
+
+
+//=========== 将预测信息（相对img_size 640）映射回原图 img0 size,对应scale_coords ============
```
Comment (wang-xinyu) [inline yolov7/common.hpp:15]: Chinese

**Review by wang-xinyu — CHANGES_REQUESTED** [review]

**Comment by lindsayshuo** [comment]
> 1. your gen_wts.py not read args -w -o but hard encode "yolov7.pt"
> 
>     2. I run -s yolov7t.wts yolov7t.engine t ,got error `Could not compute dimensions for (Unnamed Layer* 187) 

> 1. your gen_wts.py not read args -w -o but hard encode "yolov7.pt"
> 
>     2. I run -s yolov7t.wts yolov7t.engine t ,got error `Could not compute dimensions for (Unnamed Layer* 187) [Convolution]_output, because the network is not vali`



> 1. your gen_wts.py not read args -w -o but hard encode "yolov7.pt"
> 
>     2. I run -s yolov7t.wts yolov7t.engine t ,got error `Could not compute dimensions for (Unnamed Layer* 187) [Convolution]_output, because the network is not vali`

as for gen_wts.py, you can refer  to tensorrtx/yolov5;
you can use this latest code :git clone https://github.com/WongKinYiu/yolov7.git;

**Comment by lindsayshuo** [comment]
> > 1. your gen_wts.py not read args -w -o but hard encode "yolov7.pt"
> >    
> >    2. I run -s yolov7t.wts yolov7t.engine t ,got error `Could not compute dimensions for (Unnamed Layer* 187)
> 
> > 1. your gen_wts.py not read args -w -o but hard encode "yolov7.pt"
> >    
> >    2. I run -s yolov7t.wts yolov7t.engine t ,got error `Could not compute dimensions for (Unnamed Layer* 187) [Convolution]_output, because the network is not vali`
> 
> > 1. your gen_wts.py not read args -w -o but hard encode "yolov7.pt"
> >    
> >    2. I run -s yolov7t.wts yolov7t.engine t ,got error `Could not compute dimensions for (Unnamed Layer* 187) [Convolution]_output, because the network is not vali`
> 
> as for gen_wts.py, you can refer to tensorrtx/yolov5; you can use this latest code :git clone https://github.com/WongKinYiu/yolov7.git;

for yolov7.pt :  ./yolov7  -s ../yolov7.wts  ../yolov7.engine  v7

for yolov7-tiny.pt:  ./yolov7  -s ../yolov7-tiny.wts  ../yolov7.engine  t

**Comment by lindsayshuo** [comment]
> 1. your gen_wts.py not read args -w -o but hard encode "yolov7.pt"
> 
>     2. I run -s yolov7t.wts yolov7t.engine t ,got error `Could not compute dimensions for (Unnamed Layer* 187) [Convolution]_output, because the network is not vali`

for yolov7.pt : ./yolov7 -s ../yolov7.wts ../yolov7.engine v7

for yolov7-tiny.pt: ./yolov7 -s ../yolov7-tiny.wts ../yolov7.engine t

**Pull request merged**